test/dotnet.Tests/CommandTests/Project/Convert/DotnetProjectConvertTests.cs

```diff
@@ -64,6 +64,123 @@ public void SameAsTemplate()
             .And.StartWith("""<Project Sdk="Microsoft.NET.Sdk">""");
     }
 
+    [Theory] // https://github.com/dotnet/sdk/issues/50832
+    [InlineData("File", "Lib", "../Lib", "Project", "../Lib/lib.csproj")]
+    [InlineData(".", "Lib", "./Lib", "Project", "../Lib/lib.csproj")]
+    [InlineData("File", "Lib", "../Lib", "File/Project", "../../Lib/lib.csproj")]
```

Comment:
Consider also testing a case where the original path has redundant `..`s in it, such as `Lib/../Lib`, I would expect such paths to get normalized (be treated as equivalent to just `Lib`).